kasa/tests/test_discovery.py

```diff
@@ -115,7 +115,7 @@ async def test_discover_single(discovery_mock, custom_port, mocker):
     discovery_mock.port_override = custom_port
 
     device_class = Discover._get_device_class(discovery_mock.discovery_data)
-    update_mock = mocker.patch.object(device_class, "update")
+    update_mock = mocker.spy(device_class, "update")
```

Comment:
It's needed because the test needs to make sure that discovery does not call update but is changed to be `spy` instead of the previous `patch` as the `discovery_mock` already patches the `query` methods on the protocols.  I can't completely remember why I changed it, I think it was when I was verifying that the `discovery_mock` improvements work all the way down to device updates for all protocols but I've added a comment to clarify.